math/mode.go

```diff
@@ -0,0 +1,50 @@
+// mode.go
+// author(s): [CalvinNJK] (https://github.com/CalvinNJK)
+// description: Finding Mode Value In an Array
+// see mode.go
+
+package math
+
+import (
+	"errors"
+
+	"github.com/TheAlgorithms/Go/constraints"
+)
+
+// ErrEmptySlice is the error returned by functions in math package when
+// an empty slice is provided to it as argument when the function expects
+// a non-empty slice.
+var ErrEmptySlice = errors.New("empty slice provided")
+
+func Mode[T constraints.Number](numbers []T) (T, error) {
+
+	countMap := make(map[T]int)
+
+	n := len(numbers)
+
+	if n == 0 {
+		return 0, ErrEmptySlice
+	}
+
+	for i := range numbers {
```

Comment:
```suggestion
	for _, number := range numbers {
```
You don't need to declare the `temp` variable just do it here.